src/plugins/vis_type_vega/public/data_model/search_api.ts

```diff
@@ -70,7 +71,9 @@ export class SearchAPI {
           requestResponders[requestId].json(params.body);
         }
 
-        return search({ params }, { abortSignal: this.abortSignal }).pipe(
+        const searchApiParams = dataSourceId ? { params, dataSourceId } : { params };
+
+        return search(searchApiParams, { abortSignal: this.abortSignal }).pipe(
```

Comment:
How do we support imported visualization for vega?

Reply:
+1. Also I am curious about how do we support Sample Vega when we add sample data from remote data source?

Reply:
+1 for import/export

Reply:
Import visualization will be limited in support. 

If Vega visualization were to be imported from a non-MDS enabled cluster -> MDS enabled cluster, then if a datasource is specified, the associated datasource name can be added to each opensearch `url` body (which would require editing the Vega spec directly). This is what can happen when importing sample data. 

However, if the visualization were to be imported from an MDS enabled cluster -> MDS enabled cluster, then there is no current method to map new datasource names to `url` bodies. This is because the import saved objects can only determine 1 datasource. Since the Vega spec supports multiple `url` queries to multiple datasources, the import would inevitably break.

Reply:
@huyaboo import and export can be addressed in separate issue, we can check the vega script inside the saved object to see if datasource name is specified, if not, then we can add the data source name, if yes, then we don't add but checks if the data source exists. Does that simplify it?